app/src/main/java/server/ServerResponse.java

```diff
@@ -19,16 +19,48 @@ public ServerResponse(ServerRequest request, HashMap<String, Route> routeList) {
     private String returnResponse(ServerRequest requestObject){
         String routeVersion = String.valueOf(requestObject.requestLine.get("HTTPVersion"));
         String routePath = String.valueOf(requestObject.requestLine.get("Path"));
+        String routeMethod = String.valueOf(requestObject.requestLine.get("Method"));
 
-        Route obj = this.routeList.get(routePath);
+        response = this.assembleResponse(routeVersion, routePath, routeMethod);
+
+        return response;
+
+    }
+
+    private String assembleResponse(String routeVersion, String routePath, String routeMethod){
+        Route obj = this.routeMap.get(routePath);
 
         if (obj != null){
-            return obj.getObjectResponse(routeVersion, this.getHeaders(), this.currentRequest.body);
+            if (!checkIfMethodIsAllowed(this.currentRequest).equals("")){
+                return checkIfMethodIsAllowed(this.currentRequest);
+            } else{
+                String response = obj.getObjectResponse(routeVersion, this.getHeaders(), this.currentRequest.body);
+                response = trimBody(response, routePath, routeMethod);
+                return response;
+            }
         } else {
             return routeVersion +  " 404 Not Found\r\n\r\n";
         }
+    }
+
+    private String trimBody(String originalResponse, String routePath, String routeMethod){
+        if (routeMethod.equals("HEAD") || routeMethod.equals("OPTIONS")){
+            int breakIndex = originalResponse.indexOf("\r\n\r\n");
+            response = originalResponse.substring(0, breakIndex) + "\r\n\r\n";
 
+            if (routeMethod.equals("OPTIONS")){
```

Comment:
Is there a test for this code besides the ruby spec? I'm looking for a test showing that OPTIONS works for another route besides `/method_options`, for example `/simple_get`.
Same for the HEAD method. Would like to see a test confirming that HEAD works for a route like `/simple_get`.

Reply:
These have been implemented for simple_get_with_body.